src/engine/QMLUrl.js

```diff
@@ -0,0 +1,28 @@
+function QMLUrl(val) {
+  /* Return the url relative to the rootBasePath, as that will be prepended as
+   * the things that use these urls also call QMLEngine.$resolvePath. */
+  const url = `${val}`;
+
+  let resolvedPath = QmlWeb.engine.$resolvePath(url);
+  if (url === resolvedPath) {
+    return url;
+  }
+
+  if (QmlWeb.engine.$rootBasePath) {
+    resolvedPath = resolvedPath.split(/[\/\\]/);
+    const rootBasePath = QmlWeb.engine.$rootBasePath.split(/[\/\\]/);
+    let i;
+    for (i = 0; i < resolvedPath.length && i < rootBasePath.length; ++i) {
+      if (resolvedPath[i] !== rootBasePath[i]) {
+        break;
+      }
+    }
+    resolvedPath = resolvedPath.slice(i);
+    resolvedPath.unshift("");
+    return resolvedPath.join("/");
+  }
+
+  return resolvedPath;
+}
+QMLUrl.plainType = true;
```

Comment:
Argh, enumerating the methods does not work for native Qt objects. So I might have missed something.